### PR TITLE
fixes for arcane unix platforms

### PR DESCRIPTION
**Pull request by despair86**
also restores full optimisations for crypto in release mode
unit tests now pass on tier II/III systems

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -106,20 +106,17 @@ if(WITH_SHELLHOOKS)
   add_definitions(-DENABLE_SHELLHOOKS)
 endif(WITH_SHELLHOOKS)
 
-add_compile_options(${OPTIMIZE_FLAGS} ${CRYPTO_FLAGS})
+# Always build PIC
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
[the remark this answers is not in the document]
Comment (michael-loki) [reply CMakeLists.txt]: This was done in #769 too btw

File: crypto/include/sodium/crypto_int32.h [hunk 2]
```diff
@@ -1,2 +1,3 @@
+#pragma once
```
Comment (michael-loki) [inline crypto/include/sodium/crypto_int32.h]: Ideally we would include guard, pragma once can be surprising sometimes (and isn’t *as* portable)

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (michael-loki) [reply CMakeLists.txt]: Ugh typo. #784

File: crypto/include/sodium/crypto_int32.h (hunk 2 above)
Comment (jedisct1) [reply crypto/include/sodium/crypto_int32.h]: ?

**Comment by majestrate** [comment]
On Tue, Aug 27, 2019 at 07:01:23AM -0700, Frank Denis wrote:
> jedisct1 commented on this pull request.
> 
> 
> 
> > @@ -1,2 +1,3 @@
> +#pragma once
> 
> ?

probably meant to reference libsodium not the user

**Comment by jedisct1** [comment]
That file doesn't exist in libsodium, and existing headers all use guards, not `#pragma`. So I'm still a little confused. Anyway, let me know if there is anything I can do.

**Pull request merged**